controllers/consoleplugin.go

```diff
@@ -0,0 +1,405 @@
+package controllers
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	"context"
+
+	consolepluginv1 "github.com/openshift/api/console/v1alpha1"
+	pipelinesv1alpha1 "github.com/redhat-developer/gitops-operator/api/v1alpha1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	resourcev1 "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	gitopsPluginName             = "gitops-plugin"
+	displayName                  = "GitOps Plugin"
+	gitopsPluginSvcName          = gitopsPluginName + "-service"
+	proxyAlias                   = "gitops"
+	pluginImageEnv               = "GITOPS_CONSOLE_PLUGIN_IMAGE"
+	servicePort                  = 9001
+	pluginServingCertName        = "console-serving-cert"
+	kubeAppLabelApp              = "app"
+	kubeAppLabelComponent        = "app.kubernetes.io/component"
+	kubeAppLabelInstance         = "app.kubernetes.io/instance"
+	kubeAppLabelPartOf           = "app.kubernetes.io/part-of"
+	kubeAppLabelRuntimeNamespace = "app.kubernetes.io/runtime-namespace"
+	httpdConfigMapName           = "httpd-cfg"
+)
+
+func pluginDeployment() *appsv1.Deployment {
+	consolePluginImage := os.Getenv(pluginImageEnv)
+	if consolePluginImage == "" {
+		consolePluginImage = "quay.io/redhat-developer/gitops-console-plugin"
```

Comment:
This should be a variable or a constant and split into two parts

Image := quay.io/redhat-developer/gitops-console-plugin 
version := "<whatever is the version>"

Reply:
@jaideepr97 In what case we need to put default value in `common.go` ? Or just all default value should be in `common.go`?

Reply:
Yeah, I think all hard coded default values should be tracked in common.go or defaults.go

That way it is easier to make changes in the future and we don't have to hunt down where individual strings are stored